pennylane_forest/qpu.py

```diff
@@ -151,58 +158,75 @@ def __init__(
 
     def expval(self, observable):
         wires = observable.wires
-        # Single-qubit observable
-        if len(wires) == 1:
-            # identify Experiment Settings for each of the possible single-qubit observables
-            wire = wires[0]
-            qubit = self.wiring[wire]
-            d_expt_settings = {
-                "Identity": [ExperimentSetting(TensorProductState(), sI(qubit))],
-                "PauliX": [ExperimentSetting(TensorProductState(), sX(qubit))],
-                "PauliY": [ExperimentSetting(TensorProductState(), sY(qubit))],
-                "PauliZ": [ExperimentSetting(TensorProductState(), sZ(qubit))],
-                "Hadamard": [
-                    ExperimentSetting(TensorProductState(), float(np.sqrt(1 / 2)) * sX(qubit)),
-                    ExperimentSetting(TensorProductState(), float(np.sqrt(1 / 2)) * sZ(qubit)),
-                ],
-            }
-            # expectation values for single-qubit observables
-            if observable.name in ["PauliX", "PauliY", "PauliZ", "Identity", "Hadamard"]:
-                prep_prog = Program()
-                for instr in self.program.instructions:
-                    if isinstance(instr, Gate):
-                        # split gate and wires -- assumes 1q and 2q gates
-                        tup_gate_wires = instr.out().split(" ")
-                        gate = tup_gate_wires[0]
-                        str_instr = str(gate)
-                        # map wires to qubits
-                        for w in tup_gate_wires[1:]:
-                            str_instr += f" {int(w)}"
-                        prep_prog += Program(str_instr)
-
-                if self.readout_error is not None:
+
+        if observable.name == "Hermitian":
+            # <H> = \sum_i w_i p_i
+            Hkey = tuple(par[0].flatten().tolist())
+            w = self._eigs[Hkey]["eigval"]
+            return w[0] * p0 + w[1] * p1
+
+        # `measure_observables` called only when parametric compilation is turned off
+        elif not self.parametric_compilation:
```

Comment:
In which case, this can simply become `if not self.parametric_compilation:`

Reply:
Updated.